### PR TITLE
ON-1398 # `forms.generateFormUrl(formId, externalId)` to return a promise that will resolve with the same value as before

### PR DESCRIPTION
**Pull request by mymattcarroll**

### CODE REVIEW

File: docs/forms.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 |---|---|---|---|
 | `options.accessKey` | Yes | `string` | Access key provided by OneBlink. |
 | `options.secretKey` | Yes | `string` | Secret key provided by OneBlink. |
+| `options.formsRendererOrigin` | Yes | `string` | The Origin of the Forms Renderer provided by OneBlink. E.g. `https://organisation.forms.oneblink.io` |
```
Comment (kizaonline) [inline docs/forms.md]: Need update the Result section as well to include {organisation} in the form URL

**Comment by kizaonline** [comment]
Fixes https://github.com/blinkmobile/oneblink-sdk-js/issues/11

File: lib/forms.js [hunk 2]
```diff
@@ -27,39 +25,44 @@ module.exports = class Forms extends OneBlinkAPI {
...
+    return super.searchRequest(`/organisations`)
+      .then((results) => {
+        if (typeof formId !== 'number') {
+          throw new TypeError('Must supply "formId" as a number')
+        }
+        if (externalId && typeof externalId !== 'string') {
+          throw new TypeError('Must supply "externalId" as a string or not at all')
+        }
+        if (!results.organisations || !results.organisations.length) {
+          throw new Error('Must supply "externalId" as a string or not at all')
```
Comment (kizaonline) [inline lib/forms.js]: Incorrect error message

Comment (mymattcarroll) [reply lib/forms.js]: Fixed.

**Pull request merged**